fvm/src/machine/default.rs

```diff
@@ -261,6 +265,33 @@ where
         Ok(())
     }
 
+    fn commit_events(&self, events: &[StampedEvent]) -> Result<Option<Cid>> {
+        if events.is_empty() {
+            return Ok(None);
+        }
+
+        let blockstore = self.blockstore();
+
+        let amt_cid = {
+            let mut amt = Amt::new_with_bit_width(blockstore, EVENTS_AMT_BITWIDTH);
+            // TODO this can be zero-copy if the AMT supports a batch set operation that takes an
+            //  iterator of references and flushes the batch at the end.
```

Comment:
@Stebalien any quick ideas here?

Reply:
https://github.com/filecoin-project/ref-fvm/pull/1083

Reply:
Awesome, thanks! But we have one impediment to using that API as-is. Would love you hear your thoughts here: https://github.com/filecoin-project/ref-fvm/pull/1083#discussion_r1022683506.